### PR TITLE
Loki: Add $__range variable

### LINKED ISSUE #25561: Support __range_s variable in the Loki query editor.
Like with Prometheus, upcoming Loki topK queries need the correct range to return the expected number of results. $__interval is supported, but this returns more results than the limit set in the topK, which is correct but counter-intuitive and not what is needed by the user. See also the Prometheus docs here: 
https://grafana.com/docs/grafana/latest/features/datasources/prometheus/#templating

Loki is going to support the following syntax in upcoming versions:

topk(5, sum by (ipaddress) (count_over_time({filename=\"/var/log/nginx/access.log\"} | regexp \"(?P<ipaddress>\\\\d{1,3}\\\\.\\\\d{1,3}\\\\.\\\\d{1,3}\\\\.\\\\d{1,3})\"[10m])))

### LINKED ISSUE #35100: [Datasource Loki] Dynamic value for "log-range" in count_over_time function
**What would you like to be added**:
Currently it's not possible to use the selected Grafana time range (e.g. relative last 5m, 1h, .. 3d, .. etc. nor absolute from-to) in querying Loki with the function "count_over_time" e.g. using the pre-defined variable `$__range` or `$__timeFilter`. Currently only Prometheus data sources can use it as per https://grafana.com/docs/grafana/latest/variables/variable-types/global-variables/#__range

I would love to have one of those variables that can be passed to the `count_over_time` function so that dashboards using Loki as datasource can query Loki functions and visualize that without having to either hardcode relative times or provide that as an additional Dashboard Variable.

So basically my main issue is that the `count_over_time` function only accepts relative time ranges strings (e.g. 5m, 1h, 3h, .. etc.) and even those can't be referenced from a pre-defined Grafana variable.

**Why is this needed**:
Basically unleashing the power of querying Loki and visualizing that in Grafana Dashboards also for metrics related to logs / log entries and so on with full flexibility.

image

image

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
This PR adds support for `__range, __range_s and __range_ms` variables in Loki. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/25561
Fixes https://github.com/grafana/grafana/issues/35100

**Special notes for your reviewer**:
1. Run Loki query - instant and range - with `$__range` variable and see if the range variable was correctly interpolated.

### CODE REVIEW

**Review by achatterjee-grafana — COMMENTED** [review]
Minor doc change request.

**Review by achatterjee-grafana — APPROVED** [review]

File: docs/sources/datasources/loki.md [hunk 1]
```diff
@@ -154,6 +154,9 @@ provides the following functions you can use in the `Query` input field.
 | `label_values(label)`                      | Returns a list of label values for the `label`.                                      |
 | `label_values(log stream selector, label)` | Returns a list of label values for the `label` in the specified `log stream selector`.|
 
+#### Using interval and range variables
```
Comment (ivanahuckova) [inline docs/sources/datasources/loki.md:157]:
```suggestion
### Using interval and range variables
```

**Review by achatterjee-grafana — APPROVED** [review]

**Review by zoltanbedi — APPROVED** [review]
lgtm 😎

**Comment by sherifkayad** [comment]
@ivanahuckova Thanks a lot for this PR :) .. Is this already released as part of 8.06? .. Is it still going to be released later?

**Comment by ivanahuckova** [comment]
@sherifkayad It is going to be part of 8.1 release. The beta should be out soon. 🙂

**Comment by sherifkayad** [comment]
@ivanahuckova great stuff! Thanks for the info. Do we have a timeline for the 8.1 release?

**Comment by ivanahuckova** [comment]
Beta should be out this week and stable release is usually ~2 weeks after that.

**Comment by sherifkayad** [comment]
@ivanahuckova today I tried the Beta release and the `$__range` variable in the Loki count queries based dashboards works like a charm! Thanks a lot for that! Really good job getting that through! 😃

**Pull request merged**